code/datums/spells/greaterknock.dm

```diff
@@ -0,0 +1,42 @@
+/obj/effect/proc_holder/spell/aoe_turf/greaterknock
```

Comment:
Make this a subtype of the `knock` spell and you can cut a bunch of lines out of this PR

Reply:
good idea i'll see about doing that